src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,401 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 -  moving mean square kernel device helpers --------------------
+
+// calculate the position in shared memory to avoid bank conflicts
+__host__ __device__ int smem_pos(int pos)
+{
+    return pos + (pos >> 5); // since shared memory banks considered is 32
+}
+
+__device__ float square(float value)
```

Comment:
Just replace with val * val in its 2 uses. This function is not used much and could be removed